### PR TITLE
weighted loss

### PR DESCRIPTION
**Pull request by clararehmann**
adding an option to weight the loss function by 10^-N where N is the number of samples at a unique location

### CODE REVIEW

File: scripts/locator.py [hunk 1]
```diff
@@ -161,8 +162,29 @@ def sort_samples(samples):
...
+        count = 0
+        for i in range(len(sample_data)):
+            if np.isnan(sample_data.loc[i, 'x']) != True:
+                if (round(sample_data.loc[i, 'x']), round(sample_data.loc[i, 'y'])) == point:
+                    count += 1
+        loc_dict.update({point : count})
+    sample_weights = []
+    for i in range(len(sample_data)):
+        if np.isnan(sample_data.loc[i, 'x']) != True and i not in test:
+            sample_weights.append(10 ** -(loc_dict[(round(sample_data.loc[i, 'x']), round(sample_data.loc[i, 'y']))]))
```
Comment (andrewkern) [inline scripts/locator.py:186]: wondering why this is the weight @clararehmann? why not 1/n?

File: scripts/locator.py [hunk 2]
```diff
@@ -161,8 +162,29 @@ def sort_samples(samples):
...
+    uniq_locs = []
+    for i in range(len(sample_data)):
+        if np.isnan(sample_data.loc[i, 'x']) != True:
+            if (round(sample_data.loc[i, 'x']), round(sample_data.loc[i, 'y'])) not in uniq_locs:
+                uniq_locs.append((round(sample_data.loc[i, 'x']), round(sample_data.loc[i, 'y'])))
+    loc_dict = {}
+    for point in uniq_locs:
+        count = 0
+        for i in range(len(sample_data)):
+            if np.isnan(sample_data.loc[i, 'x']) != True:
```
Comment (cjbattey) [inline scripts/locator.py:179]: easier to read as `if not np.isnan(...)`

File: scripts/locator.py [hunk 3]
```diff
@@ -161,8 +162,29 @@ def sort_samples(samples):
...
+                if (round(sample_data.loc[i, 'x']), round(sample_data.loc[i, 'y'])) == point:
+                    count += 1
+        loc_dict.update({point : count})
+    sample_weights = []
+    for i in range(len(sample_data)):
+        if np.isnan(sample_data.loc[i, 'x']) != True and i not in test:
```
Comment (cjbattey) [inline scripts/locator.py:185]: here too

File: scripts/locator.py [hunk 4]
```diff
@@ -360,9 +382,13 @@ def plot_history(history,dists,gnuplot):
         ac=filter_snps(genotypes)
         checkpointer,earlystop,reducelr=load_callbacks("FULL")
         train,test,traingen,testgen,trainlocs,testlocs,pred,predgen=split_train_test(ac,locs)
+        if args.weight_samples == True:
```
Comment (cjbattey) [inline scripts/locator.py:385]: don't need the `==True`

File: scripts/locator.py [hunk 5]
```diff
@@ -91,6 +91,7 @@
                     0 = silent. 1 = progress bars for minibatches. 2 = show epochs. \
                     Yes, 1 is more verbose than 2. Blame keras. \
                     default: 1. ')
+parser.add_argument('--weight_samples',default=False,type=bool,help='normalize sample loss across space (for uneven location sampling)')
```
Comment (cjbattey) [inline scripts/locator.py:94]: Let's set this up like the other boolean arguments: 
`parser.add_argument("--weight_samples",default=False,action="store_true",
                    help="normalize sample loss across space to account for uneven location sampling.")`
Then `args.weight_samples` evaluates to `False` unless `--weight_samples` is included in the command.

**Review by cjbattey — CHANGES_REQUESTED** [review]
This looks super cool! I left some comments for minor code changes, but I also want to see some sort of validation tests before we merge to master. Can you show me some output for this? I can zoom evenings or most of the day on Tuesdays and Fridays if it's easier in person. We should probably draft a section in the readme to explain the option too.

**Review by cjbattey — APPROVED** [review]
looks good to me.

**Comment by cjbattey** [comment]
Realized this never got merged -- sorry about that. Are we still happy with this implementation @clararehmann? Any objections to merging now?
